cms/djangoapps/contentstore/api/views/course_validation.py

```diff
@@ -123,11 +148,20 @@ def _assignments_validation(self, course):
         )
 
         return dict(
-            total_number=len(assignments),
-            total_visible=len(visible_assignments),
-            num_with_dates=num_with_dates,
-            num_with_dates_after_start=num_with_dates_after_start,
-            num_with_dates_before_end=num_with_dates_before_end,
+            all=dict(
+                total_number=len(assignments),
+                total_visible=len(visible_assignments),
+                num_with_dates=num_with_dates,
+                num_with_dates_after_start=num_with_dates_after_start,
+                num_with_dates_before_end=num_with_dates_before_end,
+            ),
+            graded=dict(
+                total_number=len(graded_assignments),
+                total_visible=len(graded_visibile_assignments),
+                assignments_without_dates=graded_assignments_without_dates,
+                assignments_with_dates_before_start=graded_assignments_with_dates_before_start,
+                assignments_with_dates_after_end=graded_assignments_with_dates_after_end,
```

Comment:
Nice to have: stick with the same nomenclature and keep these consistent with what's returned for `all`.  So, instead of `assignments_with_dates_before_start` -> `num_with_dates_after_start`.   Or, if you prefer the new fields, then change what's returned in `all` so it's consistent with `graded`.

Having said that, I still prefer not returning both `all` and `graded` - unless the frontend needs both values.  Instead, pass a parameter like `include_graded_assignments_only` for example.

Reply:
That makes sense. We actually need a list of the assignments because we're displaying the assignment names and linking to them on the course outline, so I can change what's return in `all` to be consistent.

As far as the `after_start` versus `before_start` distinction, I could continue to pass in the `after_start` and `before_end` assignments, but then I'd need to to also return an array of all the assignments and then do some list subtraction on the front end to determine which assignments are outside the start and end dates. I know you mentioned that your vision is for the business logic to live in the front end. Is this an example or where we'd prefer to the front end to interpret data this way?

Reply:
Oh. I see.  I didn't notice that it was a list instead of a count.
My suggestion is to:
* Remove the `all` and `graded` distinction - just return a single payload - since you are now introducing a new `return_only_graded` URL parameter.
* Yes, feel free to change the logic here on the backend if that makes the code simpler in the frontend.  The objective was to help the team move fast and experiment/evolve the logic quickly - so updating the backend to do so meets that objective.